docroot/modules/custom/mass_alerts/src/Controller/ApiController.php

```diff
@@ -0,0 +1,229 @@
+    $this->renderer = $renderer;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('date.formatter'),
+      $container->get('renderer'),
+    );
+  }
+
+  /**
+   * Returns the sitewide alerts rendered.
+   */
+  public function handleSiteRequest(Request $request) {
+
+    $results = [];
+    $nodeStorage = $this->entityTypeManager()->getStorage('node');
+
+    // Load sitewide.
+    $query = $nodeStorage->getQuery();
+    $query->condition('field_alert_display.value', 'site_wide');
+    $query->condition('type', 'alert');
+    $query->condition('status', 1);
+
+    $sitewide = $query->execute();
+    // Last item.
+    $sitewide = reset($sitewide);
+
+    if (!empty($sitewide)) {
+      $node = $nodeStorage->load($sitewide);
+      $changed_date = $node->getChangedTime();
+      $id = $node->uuid() . '__' . $changed_date;
+      $prefix = NULL;
+      $severity = $node->get('field_alert_severity')->getString();
+
+      if ($severity == 'informational_notice') {
+        $prefix = $this->t('Informational Alert');
+      }
+
+      $emergencyAlerts = [
+        'id' => $id,
+        'buttonAlert' => [
+          'hideText' => $this->t('Hide'),
+          'showText' => $this->t('Show'),
+          'text' => $this->t('Alerts'),
+        ],
+        'emergencyHeader' => [
+          'title' => $node->label(),
+        ],
+      ];
+
+      if ($prefix) {
+        $emergencyAlerts['emergencyHeader']['prefix'] = $prefix;
+      }
+
+      $results = $emergencyAlerts;
+      $results['emergencyAlerts'] = $emergencyAlerts;
+      $results['emergencyAlerts']['alerts'] = [];
+
+      $alerts = [];
+      $alert_items = $node->get('field_alert')->referencedEntities();
+      foreach ($alert_items as $item) {
+        $item_id = $item->uuid() . '__' . $changed_date;
+        $timestamp = $item->get('field_emergency_alert_timestamp')->getString();
+        $unix_timestamp = strtotime($timestamp);
+        $timestamp = $this->dateFormatter->format($unix_timestamp, 'custom', 'M. jS, Y, h:i a');
+
+        $uri = $item->get('field_emergency_alert_link')->getString();
+        $url = Url::fromUri($uri)->toString();
+        $link_type = $item->get('field_emergency_alert_link_type')->getString();
+
+        $link = [
+          'chevron' => TRUE,
+          'href' => $url,
+          'text' => 'Read more',
+          'type' => $link_type,
+        ];
+
+        $alerts[$unix_timestamp] = [
+          'id' => $item_id,
+          'link' => $link,
+          'message' => $item->get('field_emergency_alert_message')->getString(),
+          'timeStamp' => $timestamp,
+        ];
+      }
+
+      ksort($alerts);
+      $results['emergencyAlerts']['alerts'] = array_values($alerts);
+    }
+
+    $build = [
+      '#theme' => 'mass_alerts_sitewide',
+      '#emergencyAlerts' => $results['emergencyAlerts'],
+      '#cache' => ['max-age' => 0],
```

Comment:
We need full cache metadata before we merge this. Sitewide responses expire after 60 seconds and page/org responses get no expiration (they will expire after 30 min per cloudflare policy).